packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -233,6 +235,14 @@ export class Virtualizer<T extends object, V, W> {
     return this._collection ? this._collection.getItem(key) : null;
   }
 
+  /**
+   * Set the keys of keys for layoutInfos persisted by the virtualizer.
+   * This is to prevent these views from being reused by the virtualizer.
+   */
+  set persistedKeys(persistedKeys: Set<Key>) {
+    this._persistedKeys = persistedKeys;
+  }
```

Comment:
This may need to trigger an update to the visible views, so we should call `this.updateSubviews()` here, but only if the `persistedKeys` are different from `this._persistedKeys` (which might happen due to the `useMemo` you added.

Reply:
Right, I remember you saying that, but it didn't end up in my notes.

Reply:
I made this change but didn't see a behavior difference. It worked before and after the change.

Reply:
A scenario would be if `persistedKeys` were set to include a row that isn't currently visible. Might not occur in current scenarios, but it is a possible case.